app/domain/authentication/authn_jwt/signing_key/create_signing_key_interface.rb

```diff
@@ -44,30 +47,36 @@ def validate_key_configuration
         end
 
         def provider_uri_has_valid_configuration?
-          return @provider_uri_has_valid_configuration unless @provider_uri_has_valid_configuration.nil?
+          return @provider_uri_has_valid_configuration if defined?(@provider_uri_has_valid_configuration)
+
           @provider_uri_has_valid_configuration ||= fetch_provider_uri_signing_key.valid_configuration?
         end
 
         def jwks_uri_has_valid_configuration?
-          return @jwks_uri_has_valid_configuration unless @jwks_uri_has_valid_configuration.nil?
+          return @jwks_uri_has_valid_configuration if defined?(@jwks_uri_has_valid_configuration)
+
           @jwks_uri_has_valid_configuration ||= fetch_jwks_uri_signing_key.valid_configuration?
         end
 
         def fetch_provider_uri_signing_key
-          @fetch_provider_uri_signing_key ||= @fetch_provider_uri.new(authentication_parameters: @authentication_parameters,
-                                                                      logger: Rails.logger,
-                                                                      fetch_required_secrets: Conjur::FetchRequiredSecrets.new,
-                                                                      resource_class: ::Resource,
-                                                                      discover_identity_provider: Authentication::OAuth::DiscoverIdentityProvider.new)
+          @fetch_provider_uri_signing_key ||= @fetch_provider_uri.new(
+            authentication_parameters: @authentication_parameters,
+            logger: Rails.logger,
+            fetch_required_secrets: Conjur::FetchRequiredSecrets.new,
+            resource_class: ::Resource,
+            discover_identity_provider: Authentication::OAuth::DiscoverIdentityProvider.new
+          )
         end
 
         def fetch_jwks_uri_signing_key
-          @fetch_jwks_uri_signing_key ||= @fetch_jwks_uri.new(authentication_parameters: @authentication_parameters,
-                                                              logger: Rails.logger,
-                                                              fetch_required_secrets: Conjur::FetchRequiredSecrets.new,
-                                                              resource_class: ::Resource,
-                                                              http_lib: Net::HTTP,
-                                                              create_jwks_from_http_response: CreateJwksFromHttpResponse.new)
+          @fetch_jwks_uri_signing_key ||= @fetch_jwks_uri.new(
+            authentication_parameters: @authentication_parameters,
+            logger: Rails.logger,
+            fetch_required_secrets: Conjur::FetchRequiredSecrets.new,
+            resource_class: ::Resource,
+            http_lib: Net::HTTP,
+            create_jwks_from_http_response: CreateJwksFromHttpResponse.new
```

Comment:
i'll ask again - why are these not set as dependencies in the `fetch_jwks_uri` class? they should be set in there in the init method signature with default values.

Reply:
Moved